File: 2022/04/python/part2.py
with open("../input.txt") as file:
    input = file.read()
file.close()

assignment_pairs = input.split("\n")

assignment_pairs = [x.split(',') for x in assignment_pairs]

partial_matches_amount = 0

for assignment_pair in assignment_pairs:
    first_assignment_start = int(assignment_pair[0].split('-')[0])
    first_assignment_end = int(assignment_pair[0].split('-')[1])
    second_assignment_start = int(assignment_pair[1].split('-')[0])
    second_assignment_end = int(assignment_pair[1].split('-')[1])

    if first_assignment_start >= second_assignment_start and first_assignment_start <= second_assignment_end:
        partial_matches_amount += 1

    elif second_assignment_start >= first_assignment_start and second_assignment_start <= first_assignment_end:
        partial_matches_amount += 1

print(partial_matches_amount)

File: 2022/04/javascript/part2.js
let fs = require('fs');
const input = fs.readFileSync('../input.txt', 'utf-8').toString();

let assignmentPairs = input.split('\n');

console.log(assignmentPairs);

assignmentPairs = assignmentPairs.map((assignmentPair) =>
  assignmentPair.split(',')
);

let partialMatchesAmount = 0;

for (const assignmentPair of assignmentPairs) {
  let assignment1start = +assignmentPair[0].split('-')[0];
  let assignment1end = +assignmentPair[0].split('-')[1];

  let assignment2start = +assignmentPair[1].split('-')[0];
  let assignment2end = +assignmentPair[1].split('-')[1];

  // Check if first range is fully before the second

  if (
    assignment1start < assignment2start &&
    assignment1end < assignment2start
  ) {
    console.log('no overlap');
  }

  // Check if second range is fully before the first
  else if (
    assignment2start < assignment1start &&
    assignment2end < assignment1start
  ) {
    console.log('no overlap');
  } else {
    partialMatchesAmount += 1;
  }
}

console.log(partialMatchesAmount);
